### PR TITLE
[Part1] Implement test files

### PR DESCRIPTION
**Pull request by Bodong-Yang**
## Introduction
This PR introduce some test files for otaclient and refactors/refines how the tests are performed. 
Also, this PR fix an issue of `app.common.copytree_identical` caused by unexpected behavior `os.walk`(list symlinks to dir in the <dirs> list).

NOTE: future updates will be introduced by other PRs.

## Changes
1. fix `app.common.copytree_identical`,
2. refactor Dockerfiles for ota test, now we use only one image and launch one container to carry out the test,
3. refinement over Dockerfiles to reduce the test image size,
4. implement some helper functions and fixtures in `tests.utils` and `tests.conftest`,
5. implement test files for `app.common` and `app.create_standby` modules,
6. refactor github actions, cleanup unused files.

### CODE REVIEW

File: app/common.py [hunk 1]
```diff
@@ -236,6 +240,14 @@ def copytree_identical(src: Path, dst: Path):
             dirs.clear()  # stop iterate the subfolders of this dir
             continue
 
+        # NOTE(20220803): os.walk now lists symlinks pointed to dir
+        # in the <dirs> tuple, we have to handle this behavior
+        for _dir in dirs:
+            _src_dir = _cur_dir_on_src / _dir
+            _dst_dir = _cur_dir_on_dst / _dir
+            if (not _src_dir.is_symlink()) and _dst_dir.is_symlink():
+                _dst_dir.unlink()
```
Comment (obi-t4) [inline app/common.py:249]: Is this block already done in L188?

**Review by obi-t4 — COMMENTED** [review]
Can you update DEVELOPMENT.md especially `Run tests manually`?

File: tests/test_common.py [hunk 2]
```diff
@@ -0,0 +1,276 @@
...
+    verify_file,
+    write_to_file_sync,
+)
+from tests.utils import compare_dir
+
+logger = logging.getLogger(__name__)
+
+_TEST_FILE_CONTENT = "123456789abcdefgh" * 3000
+_TEST_FILE_SHA256 = sha256(_TEST_FILE_CONTENT.encode()).hexdigest()
+_TEST_FILE_LENGTH = len(_TEST_FILE_CONTENT)
```
Comment (obi-t4) [inline tests/test_common.py:30]: Not to depend on contents.
```suggestion
_TEST_FILE_LENGTH = len(_TEST_FILE_CONTENT.encode())
```

**Comment by Bodong-Yang** [comment]
@obi-t4 Oh sorry I also include the latest commits into this branch, I planned to split the unit tests implementation into 2~3 parts and submit different PRs.

**Comment by Bodong-Yang** [comment]
> @obi-t4 Oh sorry I also include the latest commits into this branch, I planned to split the unit tests implementation into 2~3 parts and submit different PRs.

I will cleanup branches and re-submit PR.

**Review by obi-t4 — COMMENTED** [review]
Some comments

File: app/common.py (hunk 1 above)
Comment (Bodong-Yang) [reply app/common.py:249]: L188~ is iterating under the src folder, while L243~ is under through the dst folder.

File: app/common.py [hunk 3]
```diff
@@ -215,11 +212,18 @@ def copytree_identical(src: Path, dst: Path):
...
 
             # copy/symlink dst as src
             #   if src is symlink, check symlink, re-link if needed
             if _src_f.is_symlink():
-                dst_symlink_as_src(_src_f, _dst_f)
+                if not _dst_f.is_symlink() and _dst_f.is_dir():
+                    # if dst is a dir, remove it
+                    shutil.rmtree(_dst_f, ignore_errors=True)
+                else:  # dst is symlink or file
+                    _dst_f.unlink()
```
[the remark this answers is not in the document]
Comment (Bodong-Yang) [reply app/common.py:225]: Yes! L213-L216 is redundant code, I will delete these lines.

**Pull request closed without merging**